tests/k8st/utils/utils.py

```diff
@@ -19,6 +19,23 @@
 
 _log = logging.getLogger(__name__)
 
+# Helps with printing diags after a test.
+class DiagsCollector(object):
```

Comment:
I think it would be good to define an `__init__` with some args that customize what the Diag outputs.
As an example, there could be like a `dump_routes` variable to print out `docker exec kube-node-extra ip r` for service advertisement tests

Reply:
Yeah, we could add some arguments to this, but for the information I was looking for right now I didn't need any so I didn't add any :D 

Once there's a compelling reasons to pass arguments we can add them.

Dumping routes on the remote node already shows up in the log output, but adding confd template dumps is on my list of things we could do for sure.